actions.js

```diff
@@ -1,228 +1,281 @@
-                ]
-            }
-        });
-    }
+function dateToEpoch(date) {
+  return Math.floor(date.getTime() / 1000.0 * 1000000)
+}
+
+async function vestingPrivateTermination(contract, requestKind) {
+  let accountId = contract.accountId;
+  let lockupAccountId = document.querySelector('#lockup-account-id').value;
+  let lockupVestingStartDate = dateToEpoch(new Date(document.querySelector('#lockup-vesting-start-date').value));
+  let lockupVestingEndDate = dateToEpoch(new Date(document.querySelector('#lockup-vesting-end-date').value));
+  let lockupVestingCliffDate = dateToEpoch(new Date(document.querySelector('#lockup-vesting-cliff-date').value));
+  let lockupVestingSalt = document.querySelector('#lockup-vesting-salt').value;
+  if (!await utils.accountExists(window.near.connection, lockupAccountId)) {
+    alert(`Account ${lockupAccountId} doesn't exist`);
+    return;
+  }
+  const lockupAccount = await window.near.account(lockupAccountId);
+  console.log(`Vesting ${requestKind} for ${lockupAccountId}`);
+
+  let publicKeyStr = document.querySelector('#new-key').value;
+  let publicKey = nearAPI.utils.PublicKey.fromString(publicKeyStr);
+
+  const salt = Buffer.from(sha256(Buffer.from(lockupVestingSalt + publicKey)), 'hex').toString('base64');
+
+  let args = {
+    vesting_schedule_with_salt:
+      {
+        vesting_schedule:
+          {
+            start_timestamp: lockupVestingStartDate.toString(),
+            cliff_timestamp: lockupVestingCliffDate.toString(),
+            end_timestamp: lockupVestingEndDate.toString()
+          },
+        salt
+      }
+  }
+  //console.log(args);
+  //args = Buffer.from(JSON.stringify(args)).toString('base64');
+  //console.log(args);
+
+  try {
+    await contract.functionCall(accountId, 'add_request', {
+      request: {
+        receiver_id: lockupAccountId,
+        actions: [funcCall("terminate_vesting", args)],
+      }
+    });
+  } catch (e) {
+    console.log(e);
+  }
+
 }
 
 async function setupMultisig(contract) {
 
 }
 
 async function setupLockup(contract) {
-    let accountId = document.querySelector('#create-lockup-account-id').value;
-    let amount = document.querySelector('#create-lockup-amount').value;
-    let duration = document.querySelector('#create-lockup-duration').value;
-    let allowStaking = document.querySelector('#create-lockup-staking').value;
-    if (!await utils.accountExists(window.near.connection, accountId)) {
-        alert(`${accountId} doesn't exit. Create it first.`);
-        return;
-    }
-    amount = utils.parseAmount(amount);
-    try {
-        duration = parseInt(duration);
-    } catch (error) {
-        alert(`Failed to parse duration ${duration}`);
-        return;
-    }
-    // Days to nano seconds.
-    duration = duration * 60 * 60 * 24 * 1000 * 1000 * 1000;
-    await deployLockup(contract, accountId, amount, duration, allowStaking);
+  let accountId = document.querySelector('#create-lockup-account-id').value;
```

Comment:
why is everything reformatted, actually?
may be reformat if really needed in separate PR? and add a js linter than to format automatically

Reply:
Apologies, reformatting was down automatically by the linter, I'll switch it off next time to a manual mode.